mne/io/kit/kit.py

```diff
@@ -328,181 +285,268 @@ def _read_segment(self, start=0, stop=None, sel=None, verbose=None,
             data = np.fromfile(fid, dtype='h', count=count)
             data = data.reshape((buffer_size, nchan))
         # amplifier applies only to the sensor channels
-        n_sens = self._sqd_params['n_sens']
-        sensor_gain = np.copy(self._sqd_params['sensor_gain'])
+        n_sens = self._kit_info['n_sens']
+        sensor_gain = np.copy(self._kit_info['sensor_gain'])
         sensor_gain[:n_sens] = (sensor_gain[:n_sens] /
-                                self._sqd_params['amp_gain'])
+                                self._kit_info['amp_gain'])
         conv_factor = np.array((KIT.VOLTAGE_RANGE /
-                                self._sqd_params['DYNAMIC_RANGE']) *
+                                self._kit_info['DYNAMIC_RANGE']) *
                                sensor_gain, ndmin=2)
         data = conv_factor * data
         data = data.T
+
         # Create a synthetic channel
-        trig_chs = data[self._sqd_params['stim'], :]
-        if self._sqd_params['slope'] == '+':
-            trig_chs = trig_chs > self._sqd_params['stimthresh']
-        elif self._sqd_params['slope'] == '-':
-            trig_chs = trig_chs < self._sqd_params['stimthresh']
-        else:
-            raise ValueError("slope needs to be '+' or '-'")
-        trig_vals = np.array(2 ** np.arange(len(self._sqd_params['stim'])),
-                             ndmin=2).T
-        trig_chs = trig_chs * trig_vals
-        stim_ch = np.array(trig_chs.sum(axis=0), ndmin=2)
-        data = np.vstack((data, stim_ch))
+        if self._kit_info['stim'] is not None:
+            trig_chs = data[self._kit_info['stim'], :]
+            if self._kit_info['slope'] == '+':
+                trig_chs = trig_chs > self._kit_info['stimthresh']
+            elif self._kit_info['slope'] == '-':
+                trig_chs = trig_chs < self._kit_info['stimthresh']
+            else:
+                raise ValueError("slope needs to be '+' or '-'")
+            trig_vals = np.array(2 ** np.arange(len(self._kit_info['stim'])),
+                                 ndmin=2).T
+            trig_chs = trig_chs * trig_vals
+            stim_ch = np.array(trig_chs.sum(axis=0), ndmin=2)
+            data = np.vstack((data, stim_ch))
         data = data[sel]
 
         logger.info('[done]')
-        times = np.arange(start, stop) / self.info['sfreq']
+        times = np.arange(start, stop, dtype=float) / self.info['sfreq']
 
         return data, times
 
-    def _set_dig_kit(self, mrk, elp, hsp, auto_decimate=True):
-        """Add landmark points and head shape data to the RawKIT instance
 
-        Digitizer data (elp and hsp) are represented in [mm] in the Polhemus
-        ALS coordinate system.
+class EpochsKIT(EpochsArray):
```

Comment:
yeah, I see your point. EpochsArray has a `data` parameter to add the array
to the object. I guess I can assign the data directly to `._data` then?

-teon

On Sat, Apr 11, 2015 at 9:52 PM, Eric Larson <notifications@github.com>
wrote:

> In mne/io/kit/kit.py
> <https://github.com/mne-tools/mne-python/pull/1590#discussion_r28197901>:
>
> >
> > -        Digitizer data (elp and hsp) are represented in [mm] in the Polhemus
> > -        ALS coordinate system.
> > +class EpochsKIT(EpochsArray):
>
> Yeah, you wouldn't be instantiating an instance of _BaseEpochs here,
> you'd be subclassing _BaseEpohcs. Does EpochsArray have some
> functionality you need that _BaseEpochs doesn't have? This is similar to
> how all of our raw classes subclass _BaseRaw instead of e.g. RawArray or
> RawFIFF.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/1590/files#r28197901>.
>